dbcon/execplan/calpontsystemcatalog.cpp

```diff
@@ -6223,6 +6223,14 @@ void CalpontSystemCatalog::buildSysColinfomap()
   aCol.colPosition++;
   aCol.columnOID = OID_SYSCOLUMN_NEXTVALUE;
   fColinfomap[aCol.columnOID] = aCol;
+
+  aCol.colWidth = 4;
+  aCol.constraintType = NOTNULL_CONSTRAINT;
```

Comment:
```suggestion
  {
     fColinfomap[OID_SYSCOLUMN_CHARSETNUM] = ColType(4, INT, ddn, colPosition,OID_SYSCOLUMN_CHARSETNUM); 
  }

Reply:
Doesn't re-use ColType in an error-prone way and saves space.

Reply:
I am not fully convinced about this. I don't see why we should create a new ColType ctor just to address a special pattern that only exists in this function.

Reply:
Consider the case when someone forgets to set colWidth. This would cause an incorrect colWidth to be set. When you use a ctor one is forced to ship all values needed.